### DISCUSSION TITLE
"Visual Studio Code" at sign up

### DISCUSSION THREAD

**Discussion by Amerifax** [community/community#69635] [New to GitHub] [answered]
### Select Topic Area

Question

### Body

When I signed up a had three choices to select one. I selected "Visual Studio Code".  What where the other two?
I'm not sure what I should have picked. What is the difference? I'm looking to write code that works with Visual Basic. 
Bob

**Comment by jjrh92** [comment]
Can you please provide more context like where did you get those selection options? When signing up where or what service? in any case if you're looking to write code that works with VB then Visual Studio Code works good.

**Reply by Amerifax** [reply]
While in Visual Studio Code I found the following extension.
**Extension** - CoPilot for VS Code v1.3.0 (JavaScript/TypeScript development tooling)
Is this what I want to have CoPilot that works inside of Visual Studio Code?

**Reply by jjrh92** [reply]
Yes sir.

**Comment by Amerifax** [comment]
to jjrh92 - Visual Studio Code works good.
I did make that selection. Is this some thing I have to load on my Windows 10 platform? I have been going through the docs on the site. But I was hoping to fine something that could take me from the initial, where I selected Visual Studio Code, sign-in to where I could inter my code that I want checked.
Bob

**Reply by jjrh92** [reply, marked as answer]
You're saying that you want your code checked. You want to run it? You need to first start creating a new project on vscode go to file and make a new project with the visual basic extension you want to work with. I'm not sure if i understood correctly what you want to do.

**Reply by Amerifax** [reply]
i have signed up in something: enterprise. There is a 30 day trial. I am hoping to be able to try copilot. My plan is to use it to help in verifying my Nuance Dragon code.
I also hope to use it in developing my website. I'm using a product called Invision power, community forum. And I will be needing to use links on the website to help navigate people around.
So my first step is probably going to be trying new install copilot. I am done some research and found that I will need something called Visual Studio Code which is supposed to have a link to the GitHub copilot.
I'm hoping you will be able to tell me a claim firing right on all cylinders.
Bob Snow
Amerifax

**Reply by jjrh92** [reply]
Hello Bob, Once you got copilot enabled you can start right away. the vscode can be downloaded from their official site, just google visual studio code and install it.

**Reply by Amerifax** [reply]
Thanks. I'll give it a try tonight.
Bob

**Reply by Amerifax** [reply]
I got stuck at the Source Control. The following was the guide I was using.
Bob
![Git's](https://github.com/community/community/assets/53872456/9b4ef6b5-dd9a-4eaa-a4ec-d5c6f5edff10)

Connecting Visual Studio Code to GitHub Enterprise and Copilot: - Here's a step-by-step guide on how to connect Visual Studio Code to GitHub Enterprise and Copilot:

1. Install Git:
	• You need Git installed on your system to interact with GitHub repositories. Download and install Git from  https://git-scm.com/downloads.
	• Verify your Git installation by opening a command prompt and typing git --version.

2. Log in to GitHub Enterprise in VS Code:
	• Open VS Code.
	• In the bottom left corner, click on the Source Control tab.
	• If you're not signed in to any accounts, click on the Sign in to GitHub.com... text.

**INote** It wants me to download. I thought I already Had GitHub Installed.
	• Choose Sign in to Enterprise Account.
	• Enter your GitHub Enterprise server URL, username, and password.
	• Click Sign in.

3. Install the GitHub Pull Requests and Issues extension:
	• This extension lets you manage your pull requests and issues directly from VS Code.
	• Open the VS Code extension marketplace.
	• Search for GitHub Pull Requests and Issues.
	• Install the extension by clicking on the Install button.

4. Install Copilot:
	• Download and install Copilot from  https://github.com/features/copilot.
	• Follow the installation instructions provided by Copilot.

5. Configure Copilot in VS Code:
	• Open VS Code.
	• Open the Command Palette (Ctrl+Shift+P).
	• Search for Copilot: Configure.
	• Follow the on-screen instructions to configure Copilot.
	• This may involve linking your GitHub account to Copilot and granting access to your repositories.

6. Connect Copilot to your GitHub repositories:
	• Open a GitHub repository in VS Code.
	• Make sure you're logged in to the correct account in both VS Code and Copilot.
	• Copilot should automatically detect the repository and offer to connect.
	• Follow the on-screen instructions to connect Copilot to the repository.

7. Verify code with GitHub:
	• GitHub Enterprise doesn't have a built-in code verification feature like GitHub.com.
	• You can use external tools like CodeClimate or SonarQube for code verification.
	• These tools integrate with GitHub Enterprise and provide code quality reports and analysis.

Additional notes:
	1. Make sure you have the latest versions of VS Code and Copilot installed.
	2. You may need to restart VS Code after installing extensions.
	3. Consult the official documentation for detailed instructions and troubleshooting tips:
		• VS Code Git integration:  https://github.com/microsoft/vscode
		• GitHub Pull Requests and Issues extension:  https://marketplace.visualstudio.com/
		• Copilot:  https://github.com/features/copilot
		• CodeClimate:  https://codeclimate.com/
		• SonarQube:  https://sonarcloud.io/login

By following these steps, you should be able to connect Visual Studio Code to GitHub Enterprise and Copilot and have your code verified by external tools.

**Reply by Amerifax** [reply]
Im sorry for the above confusion. It's the best way that I felt I commit my soap understandable. The confusion comes in at the following"
Log in to GitHub Enterprise in VS Code:
• Open VS Code.
• In the bottom left corner, click on the Source Control tab.
• If you're not signed in to any accounts, click on the Sign in to GitHub.com... text.

I thought I was already using GitHub for Windows.
Bob

**Reply by jjrh92** [reply]
If you were already using it then it should sync automatically when you start vscode. That said give it a second check on the bottom left icon and make sure it's working. My discord id is 125683846506086400 probably would work better to help you.

**Comment by Amerifax** [comment]
Thanks Julio. I will start from there.
Bob
Germantown, WI.

**Comment by Amerifax** [comment]
Julio, you had a link, which I don't see now. Can you resend the link.
Bob

**Reply by jjrh92** [reply]
I don't remember posting a link for any of the replies. (?)
